src/proxy/http/client.rs

```diff
@@ -290,12 +233,7 @@ where
                 }
             },
             Settings::Http2 => {
-                let mut h2_builder = h2::client::Builder::default();
-                // h2 currently doesn't handle PUSH_PROMISE that well, so we just
-                // disable it for now.
-                h2_builder.enable_push(false);
-                let h2 = tower_h2::client::Connect::new(connect, h2_builder, BoxExecutor::new(executor));
-
+                let h2 = h2::Connect::new(connect, executor);
```

Comment:
similarly, this is probably fine for now. but we'll need to be able to configure h2 clients at some point.